dlib/dnn/tensor.h

```diff
@@ -377,7 +377,7 @@ namespace dlib
         }
 
         void set_size(
-            long n_, long k_ = 1, long nr_ = 1, long nc_ = 1
+            long long n_, long long k_ = 1, long long nr_ = 1, long long nc_ = 1
```

Comment:
You missed a (long) cast a few lines down.

Reply:
Oops, fixed.